pkg/asset/installconfig/aws/validation.go

```
@@ -166,6 +184,167 @@ func validateAMI(ctx context.Context, config *types.InstallConfig) field.ErrorLi
+			return fmt.Errorf("invalid allocation ID format (missing prefix eipalloc-) for %s: eipAllocations should start with 'eipalloc-'", id)
+		} else if !eipRegex.MatchString(id) {
+			// Rule 5: Check if the value matches the regex
+			return fmt.Errorf("invalid allocation ID format (invalid characters) for %s : eipAllocations must be 'eipalloc-' followed by exactly 17 hexadecimal characters (0-9, a-f, A-F)", id)
+		}
+	}
+	return nil
+}
+
+// Checks if the type was set to NLB when eipAllocations are provided to the install-config.
+func checkForEIPAllocationsWithNLB(config *types.InstallConfig) error {
+	if (len(config.Platform.AWS.LBType) == 0 && config.Platform.AWS.EIPAllocations != nil && len(config.Platform.AWS.EIPAllocations.IngressNetworkLoadBalancer) != 0) || (len(config.Platform.AWS.LBType) != 0 && config.Platform.AWS.LBType != configv1.NLB && config.Platform.AWS.EIPAllocations != nil && len(config.Platform.AWS.EIPAllocations.IngressNetworkLoadBalancer) != 0) {
+		return fmt.Errorf("eipAllocations can be provided only when lb type is NLB")
+	}
+	return nil
+}
+
+func validateEIPAllocations(client ec2iface.EC2API, ctx context.Context, meta *Metadata, fldPath *field.Path, config *types.InstallConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if config.Platform.AWS.EIPAllocations != nil && len(config.Platform.AWS.EIPAllocations.IngressNetworkLoadBalancer) == 0 {
+		return nil
+	}
+	var eipAllocations []awstypes.EIPAllocation
+	if config.Platform.AWS != nil && config.Platform.AWS.EIPAllocations != nil && len(config.Platform.AWS.EIPAllocations.IngressNetworkLoadBalancer) == 0 {
+		eipAllocations = config.Platform.AWS.EIPAllocations.IngressNetworkLoadBalancer
+	}
+	var allocationIDs []string
+	for _, allocationID := range eipAllocations {
+		allocationIDs = append(allocationIDs, string(allocationID))
+	}
+
+	// Validate EIP allocation IDs
+	if err := validateEIPAllocationsList(allocationIDs); err != nil {
+		fmt.Printf("Validation error: %v\n", err)
```

Do we want to return an error here?

---

yes done